firefox-ios/Client/Frontend/Home/Homepage Rebuild/TopSites/TopSitesManager.swift

```diff
@@ -72,19 +104,42 @@ class TopSitesManager {
             }
         }
 
-        return contiles.compactMap {
-            TopSiteState(site: SponsoredTile(contile: $0))
-        }
+        return contiles
+            .prefix(maxNumberOfSponsoredTile)
+            .compactMap { SponsoredTile(contile: $0) }
     }
 
-    // MARK: History-based (Frencency) + Pinned + Default suggested tiles
-    private func getHistoryBasedSites() async -> [TopSiteState] {
-        let frecencySites = await withCheckedContinuation { continuation in
+    /// Show the next sponsored site if site is already present in the pinned sites
+    /// or if it's the default search engine
+    private func shouldShowSponsoredSite(with site: Site) -> Bool {
+        // TODO: FXIOS-10477 - To replace with proper logic in when to show sponsored sites 
+        // such as if there's duplicates or if default engine is showed
+        return true
+    }
+
+    // MARK: Other Sites = History-based (Frencency) + Pinned + Default suggested tiles
+    private func getOtherSites() async -> [TopSiteState] {
+        let otherSites = await withCheckedContinuation { continuation in
             topSiteHistoryManager.getTopSites { sites in
                 continuation.resume(returning: sites)
             }
         }
 
-        return frecencySites?.compactMap { TopSiteState(site: $0) } ?? []
+        return otherSites?.compactMap { TopSiteState(site: $0) } ?? []
+    }
+
+    // MARK: - Tiles space calculation
+
+    /// Get available space count for the sponsored tiles and Google tiles, pinned tiles are prioritized first
+    /// - Parameter otherSites: Comes from fetching the other top sites that are not sponsored or google tile
+    /// - Returns: The available space count for the rest of the calculation
+    private func getAvailableSpaceCount(with otherSites: [TopSiteState]) -> Int {
+        let pinnedSiteCount = otherSites.filter { $0.site is PinnedSite }.count
+        return maxTopSites - pinnedSiteCount
```

Comment:
I am copying over the same logic that we have in our top site adaptor (see code below). I don't like how we don't check that `maxTopSites > pinnedSiteCount`, and its safe right now because our number of top sites fetched is hardcoded to always be smaller than `maxTopSites`, so I kept it as is. 

Ideally, I would've like to change our calculation and separate out the pinned sites from the other top sites, but that would cause more changes to `topSiteHistoryManager` which will increase scope, so making minimal changes if I can.

Previous implementation:
```swift
private func getAvailableSpaceCount(maxTopSites: Int) -> Int {
        let pinnedSiteCount = countPinnedSites(sites: historySites)
        return maxTopSites - pinnedSiteCount
    }
    
private func countPinnedSites(sites: [Site]) -> Int {
        var pinnedSites = 0
        sites.forEach {
            if $0 as? PinnedSite != nil { pinnedSites += 1 }
        }
        return pinnedSites
    }
 ```